experimenter/experimenter/nimbus-ui/src/components/Summary/index.tsx

```diff
@@ -38,6 +44,51 @@ const Summary = ({ experiment, refetch }: SummaryProps) => {
   const takeawaysProps = useTakeaways(experiment, refetch);
   const qaProps = useQA(experiment, refetch);
   const status = getStatus(experiment);
+  const [legalSignoff, setLegalSignoff] = useState(
+    experiment.legalSignoff || false,
+  );
+  const [qaSignoff, setQaSignoff] = useState(experiment.qaSignoff || false);
+  const [vpSignoff, setVpSignoff] = useState(experiment.vpSignoff || false);
+  const [updateExperiment] = useMutation<
+    updateExperiment,
+    updateExperimentVariables
+  >(UPDATE_EXPERIMENT_MUTATION);
+
+  const handleLegalSignoffChange = (value: boolean) => {
+    setLegalSignoff(value);
+    updateExperimentSignoff(value, qaSignoff, vpSignoff);
+  };
```

Comment:
This really *should* only propagate the state if it was successful.

```suggestion
  const handleLegalSignoffChange = (value: boolean) => {
    updateExperimentSignoff(value, qaSignoff, vpSignoff)
        .then(() => setLegalSignoff(value));
  };
```

Is it the unit tests giving you trouble?

Reply:
yup changing to this leads to error

Reply:
That is because you were using userEvent without awaiting. I just verified locally that it works if you await. Please apply this change and the suggestsions to the tests. I have verified locally that tests pass.